vcd/resource_vcd_vapp.go

```diff
@@ -17,61 +17,65 @@ func resourceVcdVApp() *schema.Resource {
 		Delete: resourceVcdVAppDelete,
 
 		Schema: map[string]*schema.Schema{
-			"name": &schema.Schema{
+			"name": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
 
-			"template_name": &schema.Schema{
+			"template_name": {
 				Type:     schema.TypeString,
-				Optional: true,
+				Required: true,
 				ForceNew: true,
 			},
 
-			"catalog_name": &schema.Schema{
-				Type:     schema.TypeString,
-				Optional: true,
-			},
-
-			"network_href": &schema.Schema{
```

Comment:
I just copy + pasted my changes I did last time against terraform repo. I see that it's now used in a different way. Have to check again if the error I had before still exists. Thanks for pointing this out!